### ISSUE TITLE
ProtoFlux node to do a "Ballistic Raycast"

### ISSUE THREAD

**Issue by JackTheFoxOtter** [Yellow-Dog-Man/Resonite-Issues#2899]
### Is your feature request related to a problem? Please describe.

The TeleportLocomotion component does a sort of "ballistic raycast" while the user is targeting where to teleport. This functionality could be pretty useful for a variety of game mechanics, like choosing where to throw an object affected by gravity. Currently, this functionality only exists within the TeleportLocomotion though, which means creators would have to implement this functionality themselves.

### Describe the solution you'd like

A ProtoFlux component that performans such a "ballistic raycast" for a given initial velocity, gravity and drag, to match the inputs of the BallisticPathmesh and TrajectoryPosition node. It should return the hit information of the first raycast hit along the ballistic path.

### Describe alternatives you've considered

Implementing this functionality in ProtoFlux manually, which works, but requires a bit of extra setup. I think a dedicated node for this could be useful, and since the functionality already exists in a different component I figure it shouldn't be too much work to implement.

**Comment by Frooxius** [comment]
I don't know if we'd add an explicit Raycast node for this - there's lots of ways you might want to do raycasts, along a number of paths, but all they'd have to do is essentially just do a series of smaller raycasts, to trace the path bit by bit.

That's something that we'd probably just let you do yourself - there's lots of ways to do and parametrize ballistic path and plethora of other paths you might want to raycast - we can't add nodes for every single one of these.

What I think is better here is once we have Nested node support, you can build your own versions that suit your needs instead.

**Comment by JackTheFoxOtter** [comment]
Yeah that's a good point. My thinking was that this would make it easier to get additional use out of the BallisticPathMesh in user creations, so an official node could make sense, as it would get more use out of an existing feature. The math isn't too complicated to implement, I could also just make a custom reusable node once we have nested node support. If you don't think it would fit in as an official node you can close this!
